gadget/mkfs.go

```diff
@@ -0,0 +1,121 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package gadget
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+// MkfsExt4 creates an EXT4 filesystem in given image file, with an optional
+// filesystem label, and populates it with the contents of provided root
+// directory.
+func MkfsExt4(img, label, contentsRootDir string) error {
```

Comment:
(Thinking aloud) This is probably overdoing it but I wonder if putting this into `gadget/mkfs/mkfs.go` would make sense. Then it would read `mkfs.Ext4(...)` and mkfs.VFat(...)` which has a nice ring to it (and is close to how its layed out on the filesystem mkfs.ext4, mkfs.vfat). But probably silly (also easy to tweak later :)

Reply:
Wonder if it would be beneficial to put all image building pieces under a separate package. We have the mkfs wrappers, parted/sfdisk wrapper and offset writer that are needed only in such scenarios.